### PR TITLE
🧪 fix sanitizeId regex and improve edge case tests

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** Identified and fixed a bug in the `sanitizeId` regular expression that incorrectly allowed consecutive hyphens (e.g., `ab--cd`), which contradicted the intended validation rules. Also addressed a testing gap by adding comprehensive positive and negative boundary tests for the `sanitizeId` logic.
📊 **Coverage:** Added explicit edge cases for valid inputs (single chars, inner hyphens) and invalid inputs (empty strings, starting/ending with hyphens, uppercase chars, underscores, and consecutive hyphens). 
✨ **Result:** Improved test coverage significantly for edge cases surrounding identifier validation. The fix ensures that identifiers used to build safe URLs are securely and correctly sanitized, preventing malformed paths or vulnerabilities.

---
*PR created automatically by Jules for task [6888643107924703557](https://jules.google.com/task/6888643107924703557) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

このPRは `sanitizeId` の正規表現バグ（連続ハイフン `ab--cd` が許可されていた）を修正し、`null`/`undefined` の型ガードを追加。あわせてエッジケースを網羅したテストを拡充しています。

- **正規表現の変更**: 旧 `/^[a-z0-9][a-z0-9-]*[a-z0-9]$|^[a-z0-9]$/` → 新 `/^[a-z0-9]+(?:-[a-z0-9]+)*$/` — 連続ハイフン・先頭末尾ハイフン・空文字列をすべて正しく拒否する。
- **型ガードの追加**: `typeof value !== "string"` チェックで `null`/`undefined` が渡された場合も安全に `Error` をスローする。
- **テスト拡充**: 有効系（単一文字、内側ハイフン）・無効系（空文字、先頭末尾ハイフン、大文字、アンダースコア、連続ハイフン、null/undefined）の8ケースを追加。
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

正規表現のバグ修正とテスト追加のみで、既存の動作を壊すリスクはありません。

新しい正規表現 `/^[a-z0-9]+(?:-[a-z0-9]+)*$/` はすべてのエッジケース（連続ハイフン・先頭末尾ハイフン・空文字・大文字・アンダースコア）を正しく拒否し、有効な識別子はそのまま通過させる。`null`/`undefined` ガードも正しく機能しており、追加されたテストケースも整合しています。変更範囲はこの2ファイルのみで影響範囲が限定的です。

特に注意が必要なファイルはありません。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/web/src/lib/sanitization.ts | 正規表現を `/^[a-z0-9]+(?:-[a-z0-9]+)*$/` に変更し、連続ハイフンのバグを修正。`null`/`undefined` の型ガードも追加。 |
| apps/web/src/lib/__tests__/sanitization.test.ts | 有効・無効識別子のエッジケースを網羅的に追加。`null`/`undefined` のランタイムガードテストも含む。 |

</details>

<h3>Flowchart</h3>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["sanitizeId(value)"] --> B{typeof value === 'string'?}
    B -- No --> C["throw Error('Invalid identifier: ...')"]
    B -- Yes --> D{"/^[a-z0-9]+(?:-[a-z0-9]+)*$/.test(value)"}
    D -- Match --> E["return value ✅"]
    D -- No match --> F["throw Error('Invalid identifier: ...')"]

    style C fill:#f66,color:#fff
    style F fill:#f66,color:#fff
    style E fill:#6a6,color:#fff
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
flowchart TD
    A["sanitizeId(value)"] --> B{typeof value === 'string'?}
    B -- No --> C["throw Error('Invalid identifier: ...')"]
    B -- Yes --> D{"/^[a-z0-9]+(?:-[a-z0-9]+)*$/.test(value)"}
    D -- Match --> E["return value ✅"]
    D -- No match --> F["throw Error('Invalid identifier: ...')"]

    style C fill:#f66,color:#fff
    style F fill:#f66,color:#fff
    style E fill:#6a6,color:#fff
```

</a>

<sub>Reviews (2): Last reviewed commit: ["test: harden sanitizeId runtime checks"](https://github.com/hiroki-org/openshelf/commit/69bff4c3983793ca06e995ac5130bcaac9f1fa74) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=40371656)</sub>

### CODE REVIEW

File: apps/web/src/lib/sanitization.ts [hunk 1]
```diff
@@ -4,8 +4,9 @@
  * Does not allow starting/ending with a hyphen or consecutive hyphens.
  */
 export function sanitizeId(value: string): string {
-  // Regex from instruction.md: /^[a-z0-9][a-z0-9-]*[a-z0-9]$|^[a-z0-9]$/
-  const slugRegex = /^[a-z0-9][a-z0-9-]*[a-z0-9]$|^[a-z0-9]$/;
+  // Allowed characters: a-z, 0-9, and hyphen.
+  // Does not allow starting/ending with a hyphen, empty string, or consecutive hyphens.
+  const slugRegex = /^[a-z0-9]+(?:-[a-z0-9]+)*$/;
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/web/src/lib/sanitization.ts:13]: 対応しました: sanitizeId に typeof value !== "string" の実行時ガードを追加し、RegExp.test の暗黙文字列化で undefined/null が通らないようにしました。検証: npm run test -- apps/web/src/lib/__tests__/sanitization.test.ts pass.

**Comment by is0692vs** [comment]
@greptile review
